android/app/src/main/java/org/fossasia/openevent/activities/SpeakerDetailsActivity.java

```diff
@@ -239,21 +247,34 @@ else if (!TextUtils.isEmpty(selectedSpeaker.getOrganisation()))
 
     private void hideEmptyURLButtons() {
 
+        int hidden = 0;
+
         if (TextUtils.isEmpty(selectedSpeaker.getLinkedin())) {
             linkedin.setVisibility(View.GONE);
+            hidden++;
```

Comment:
Instead, please use a flag with a default value of false here which gets set to true if there is even a single visible URL.

Reply:
@the-dagger 
Using a flag would mean either:
1. I have to add else with each and every ```if``` to mark the flag as true.
2. I have to add an if statement in the checking checking if atleast one of the URL is visible and mark the flag as true in that case.
In my view using an int counter like i have done or going for the second option is good.

Reply:
Set the marker to false by default and in each if block, set it's value to true.
At the end, check if it is true to show or false to hide the card.

Makes sense?

Reply:
Sorry but it doesn't.
We are actually hiding things(not showing) in the ```if``` block. Let's say one block of ```if```` runs, then that means one of the social media info is **not** available.
If we need to hide the card, we need to be sure that all the five social media info's are not available i.e. all the five ```if``` blocks need to run. If any one of them doesn't run,then we are sure to have a social media block for the speaker.

Reply:
Ok, I seem to have missed that part.
No worries, then make the visibility as GONE by default and modify the if blocks to set the visibility to VISIBLE in case that social media field exists.

Let's not hardcode the value to 5 as it's been done here :)

Reply:
Yeah that will do :)

Reply:
👍